modules/apps/callbacks/ibc_middleware.go

```diff
@@ -281,11 +282,15 @@ func (IBCMiddleware) processCallback(
 			if callbackType == types.CallbackTypeSendPacket {
 				panic(r)
 			}
+			err = errorsmod.Wrapf(types.ErrCallbackPanic, "ibc %s callback panicked with: %v", callbackType, r)
```

Comment:
trying to wrap my head around this err assignment. It's possible because an empty error initialized at line 272, I'm unsure as to what is happening with this error. If an error is returned on line 304, is this assignment overriding the return value?

Reply:
It shouldn't be, right? because `r == nil`?

Reply:
Yeah, if execution reaches line 304, that means there were no panics, thus `r == nil`

Reply:
right of course 👍